bigtable-dataflow-import/src/test/java/com/google/cloud/bigtable/dataflowimport/ImportIntegrationTest.java

```diff
@@ -0,0 +1,333 @@
+    Set<? extends Cell> expected = getTestData();
+    SequenceFileIoUtils.createFileWithData(sequenceFile, expected);
+
+    HBaseImportOptions options = createImportOptions(
+        sequenceFile.getPath(), tableId, false /** isHBase094Format **/);
+    try (BigtableTableUtils tableUtils = tableUtilsFactory.createBigtableTableUtils(
+        tableId, Bytes.toString(CF))) {
+      tableUtils.createEmptyTable();
+      assertEquals(Collections.EMPTY_SET, tableUtils.readAllCellsFromTable());
+      doImport(options);
+      assertCellSetsEqual(expected, tableUtils.readAllCellsFromTable());
+    }
+  }
+
+  /**
+   * Verifies that a Sequence File on Google Cloud Storage (GCS) can be imported
+   * into Google Cloud Bigtable.
+   *
+   * <p>This method creates a local Sequence File, copies it to the stagingLocation folder
+   * on GCS, then imports the GCS file. The GCS file is not removed by post-test cleanup.
+   */
+  @Test
+  public void testImportSequenceFile_fromGcs() throws Exception {
+    runImportFileTestFromGcs(false /** isHBase094Format **/);
+  }
+
+  @Test
+  public void testImportSequenceFile_fromGcsWithHBase094Format() throws Exception {
+    runImportFileTestFromGcs(true /** isHBase094Format **/);
+  }
+
+  private void runImportFileTestFromGcs(boolean isHBase094Format) throws Exception {
+    String tableId = "test_" + UUID.randomUUID().toString();
+    // For this test input files already exist on GCS. getTestData() returns
+    // all data that we know are in the file. See java doc for that method for more information.
+    Set<? extends Cell> expected = getTestData();
+
+    HBaseImportOptions options = createImportOptions(
+        getCloudTestDataFolder(isHBase094Format), tableId, isHBase094Format);
+    options.setFilePattern(options.getFilePattern());
+    try (BigtableTableUtils tableUtils = tableUtilsFactory.createBigtableTableUtils(
+        tableId, Bytes.toString(CF))) {
+      tableUtils.createEmptyTable();
+      assertEquals(Collections.EMPTY_SET, tableUtils.readAllCellsFromTable());
+      doImport(options);
+      assertCellSetsEqual(expected, tableUtils.readAllCellsFromTable());
+    }
+  }
+
+  @Test
+  public void testImportSequenceFile_currentFormatWithHBase094Deserializer() throws Exception {
+    String tableId = "test_" + UUID.randomUUID().toString();
+    File sequenceFile = tmpFolder.newFile(tableId);
+    SequenceFileIoUtils.createFileWithData(sequenceFile, getTestData());
+
+    HBaseImportOptions options = createImportOptions(
+        sequenceFile.getPath(), tableId, true /** isHBase094Format **/);
+    options.setFilePattern(options.getFilePattern());
+    try (BigtableTableUtils tableUtils = tableUtilsFactory.createBigtableTableUtils(
+        tableId, Bytes.toString(CF))) {
+      assertFalse(tableUtils.isTableExists());
+      tableUtils.createEmptyTable();
+      try {
+        doImport(options);
+        fail("Expecting RuntimeException");
+      } catch (RuntimeException expected) {}
```

Comment:
Type of error depends on actual data and I'm not 100% certain I got the complet set. 
I'll add some comments to explain.